spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1121,6 +1121,18 @@ def do_completion(self, automatic=False):
     @handles(LSPRequestTypes.DOCUMENT_COMPLETION)
     def process_completion(self, params):
         """Handle completion response."""
+
+        def filter_completion(completion, include_fallback):
+            # Remove empty completion entries
+            info_condition = (completion.get('insertText')
+                              or completion.get('textEdit', {}).get('newText'))
+            # Discard fallback entries if they were not requested explicitly
+            fallback_condition = (not automatic
+                                  or completion['provider'] != 'Fallback')

```

Comment:
We should be probably use `CONSTANTS` instead of  strings,

Just making code up, to illustrate the idea.

```python
completion['provider'] != providers.FALLBACK
```